AzureFunctions.AngularClient/src/app/shared/services/site.service.ts

```
@@ -0,0 +1,62 @@
+import { Injectable, Injector } from "@angular/core";
+import { ConditionalHttpClient } from "app/shared/conditional-http-client";
+import { UserService } from "app/shared/services/user.service";
+import { CacheService } from "app/shared/services/cache.service";
+import {AuthzService} from "./authz.service";
+import { ArmSiteDescriptor } from "app/shared/resourceDescriptors";
+import { Observable } from "rxjs/Observable";
+import { HttpResult } from './../models/http-result';
+import { ArmObj } from "app/shared/models/arm/arm-obj";
+import { Site } from "app/shared/models/arm/site";
+
+@Injectable()
+export class SiteService {
+    private readonly _client: ConditionalHttpClient;
+
+    constructor(
+        userService: UserService,
+        injector: Injector,
+        private _cacheService: CacheService) {
+
+        this._client = new ConditionalHttpClient(injector, _ => userService.getStartupInfo().map(i => i.token))
+    }
+
+    getSite(resourceId: string): Observable<HttpResult<ArmObj<Site>>>{
+        const getSite = this._cacheService.getArm(resourceId).map(r => r.json());
+        return this._client.executeWithConditions(
+            ['HasPermissions'],
+            {
+                resourceId: resourceId,
+                permissionsToCheck: [AuthzService.readScope]
+            },
+            t => getSite);
+    }
+
+    getAppSettings(resourceId: string){
+        const getAppSettings = this._cacheService.postArm(`${resourceId}/config/appSettings/list`, true)
+        .map(r =>r.json());
+
+        return this._client.executeWithConditions(
+            ['HasPermissions', 'NoReadonlyLock'],
```

>['HasPermissions', 'NoReadonlyLock'], [](start = 12, length = 37)

Actually thinking about this some more, maybe it doesn't make sense to do precondition checks for ARM requests if we're not going to fail the observable.  I need to double check what kind of errors we get back but if they're reliable, then perhaps we should just do the requests, let them fail, and handle them properly.  That would also be faster in terms of performance since RBAC checks can sometimes be slow.

---

Okay yeah.  So ARM gives us pretty good errors for these situations.  I think I'm going to change this so that we just make the request and fail gracefully with distinct errors.

---
In reply to: [164780673](https://github.com/Azure/azure-functions-ux/pull/2266#discussion_r164780673) [](ancestors = 164780673)